docs/proposals/00x-epp-compliance-proposal/README.md

```diff
@@ -0,0 +1,101 @@
+
+The other layers are handled in sequential process. Starting with the **Ext-Proc** call. The request is buffered and then sent to the **Routing Layer**, which processes any User defined per-InferenceModel routing rules & request enrichment happening first (at the time of writing that is currently just translating the InferenceModel name to a weight-split actual model). Then _all_ requests pass through the **Flow Controller** to ensure that any request entry to the pool adhereing to the guidelines set by the Priority,Fairness, & Queueing configuration. And finally, the **Scheduling Layer** is the load balancing algorithm that intelligently routes requests based on the current state of the InferencePool.
+
+## Components
+
+To further expand upon these component layers. We will first break them into `extensible` and `non-extensible` layers. `Non-extensible` layers are intended to be static, and handled on behalf of the user, typically implementing low-opinion infrastructure. 
+
+The `Extensible` layers are:
+- Routing Layer
+- Scheduling Layer
+- Data Layer
+- Flow Controller
+
+The `Non-Extensible` layer(s)s are:
+- The Ext-Proc Server
+
+### `Extensible`
+
+#### Routing Layer
+
+The routing layer is likely to be the most opinion heavy section, as the scope of what constitutes a 'Route Rule' is somewhat broad. The current examples we expect would be:
+
+- System Prompt injection
+- RAG callout
+- Per-InferenceModel request validation (such as saftey/on-topic, etc)
+
+Due to the possibility of this becoming a bit of a dumping ground. The API will keep a _very_ tight scope on which of these route rules are included in the spec. A standard method of extension will be provided if the need to define a custom rule arises.
+
+#### Scheduling Layer
+
+As the Scheduling Layer is the final interface to the entirety of the pool, all configuration will be at the _pool_ level. The default scheduling layer will be an experimentally-backed LB algorithm, with exposed config values.
+
+The Scheduler will define a strong interface API, so that new scheduling algos may be plugged & dark-launched to test in production traffic without impacting said traffic. Extension is expected to adhere to the [Scheduler Subsystem definition](https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/603)
+
+
+#### Data Layer
+
+The data layer will consume and store: the InferencePool/InferenceModel config and the pre-defined [Model Server Protocol](../003-model-server-protocol/README.md) metrics. Additionally, the model server metrics will be processed and digested to provide resource scarcity regime hints, and autoscaling reccomendations.
+
+Many extensions to scheduling will require changes to ingested metrics, as such, the data layer will be built to be extended, but extenders accept that the Model Server Protocol will no longer provide guarantees on portability of a model server out of the box. 
+
```

Comment:
data layer may be extended to collect more than just metrics. for example, we can scrape from pods the models using `/v1/models` as suggested in [this issue](#715).